### PR TITLE
Implement Login Page

### PR DESCRIPTION
**Pull request by cxfloeder**
I updated the Login servlet to pass back a login/logout URL. I created a login.js file that loads the login URL when the user clicks submit on the login page. After they login, the user is automatically taken through the OAuth process. When the OAuth process is done, the user is redirected to the read page.

TODO: Possibly add a logout button. Not necessarily that important though.

### CODE REVIEW

File: src/main/java/servlets/LoginServlet.java [hunk 1]
```diff
@@ -13,27 +13,56 @@
...
+    loginObject.put("loginURL", loginURL);
+    loginObject.put("logoutURL", logoutURL);
+    loginObject.put("loginStatus", loginStatus);
+
+    // Convert the URL to a JSON String.
+    String jsonMessage = messageListAsJson(loginObject);
+
+    // Send the JSON message as the response.
+    response.setContentType(TEXT_TYPE);
+    response.getWriter().println(jsonMessage);
```
Comment (cadecrandall) [inline src/main/java/servlets/LoginServlet.java:59]: We have a utility class and response builder that already passed a login/logout URL. Can you refactor to use these classes or delete them?

File: src/main/webapp/index.html [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <meta http-equiv="refresh" content="0; url=readfile.html" />
+    <meta http-equiv="refresh" content="0; url=login.html" />
   </head>
   <body>
   </body>
```
Comment (cadecrandall) [inline src/main/webapp/index.html:7]: IMO it makes more sense to have the login page on index.html rather then sending a redirect

Comment (frostyclawz) [reply src/main/webapp/index.html:7]: Either way, it could work. We just need an index.html

File: src/main/webapp/scripts/login.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+async function loadLogInURL() {
```
Comment (frostyclawz) [inline src/main/webapp/scripts/login.js:1]: Could give a clarifying comment to thins function?

File: src/main/webapp/index.html (hunk 2 above)
Comment (cxfloeder) [reply src/main/webapp/index.html:7]: Cool, I'll move login to index.html.

File: src/main/webapp/scripts/login.js (hunk 3 above)
Comment (cxfloeder) [reply src/main/webapp/scripts/login.js:2]: Yes, if the user isn't logged in, they will be redirected to the login URL. If they are logged in, they will be sent to the readfile.html page.

File: src/main/java/servlets/LoginServlet.java [hunk 4]
```diff
@@ -13,27 +13,56 @@
...
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType(CONTENT_TYPE_TEXT_HTML);
-    response.getWriter().println(ResponseBuilder.toJson(UserAuthUtil.isUserLoggedIn(), REDIRECT_LINK));
+    response.setContentType(TEXT_TYPE);
+    UserService userService = UserServiceFactory.getUserService();
+    HashMap<String, String> loginObject = new HashMap<String, String>();
+    String loginURL = "";
+    String logoutURL = "";
+    String loginStatus = "";
```
Comment (ianjjohnson) [inline src/main/java/servlets/LoginServlet.java:40]: Why store these as local variables if they're just going to get copied into the loginObject?

I think you can avoid this entirely by just replacing lines 43, 44, 46,  47 with loginObject.put(....)

Also, use constants for all of these strings

Comment (cxfloeder) [reply src/main/java/servlets/LoginServlet.java]: Will do

**Pull request merged**